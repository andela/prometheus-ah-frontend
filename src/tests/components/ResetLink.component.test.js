import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ResetPassword from '../../components/ResetLink';
import mockData from '../__mocks__/mockData';


let resetPasswordRequest;
let wrapper;

const createMockStore = configureMockStore([thunk]);

describe('Reset Email Link', () => {
  beforeEach(() => {
    const { email } = mockData.userEmail;
    const store = createMockStore({});
    resetPasswordRequest = jest.fn();
    wrapper = shallow(
      <ResetPassword.WrappedComponent
        resetPasswordRequest={resetPasswordRequest}
        resetEmail={email}
        store={store}
      />
    );
  });

  test('should correctly render Reset Email link ', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should call resetPasswordRequest on click of the link', () => {
    const { email } = mockData.userEmail;
    wrapper.find('.alert-link').simulate('click', {
      preventDefault: () => {}
    });
    expect(resetPasswordRequest).toHaveBeenCalledWith(email);
    expect(wrapper).toMatchSnapshot();
  });
});
