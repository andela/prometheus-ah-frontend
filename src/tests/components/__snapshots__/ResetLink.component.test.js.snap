// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Email Link should call resetPasswordRequest on click of the link 1`] = `
<div>
  <div>
    <div
      className="alert alert-success alert-dismissible fade show reset-alert"
      role="alert"
    >
      <p
        className="mb-0"
      >
        A password reset link has been sent to test@mymail.com, Click
         
        <Link
          className="alert-link"
          onClick={[Function]}
          replace={false}
          to="#"
        >
          Here
        </Link>
         
        to get a new password reset link
      </p>
      <button
        aria-label="Close"
        className="close"
        data-dismiss="alert"
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Reset Email Link should correctly render Reset Email link  1`] = `
<div>
  <div>
    <div
      className="alert alert-success alert-dismissible fade show reset-alert"
      role="alert"
    >
      <p
        className="mb-0"
      >
        A password reset link has been sent to test@mymail.com, Click
         
        <Link
          className="alert-link"
          onClick={[Function]}
          replace={false}
          to="#"
        >
          Here
        </Link>
         
        to get a new password reset link
      </p>
      <button
        aria-label="Close"
        className="close"
        data-dismiss="alert"
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
      </button>
    </div>
  </div>
</div>
`;
